### ISSUE TITLE
Unable to connect to Workgroup database for "What's Going On" TabMon dashboard

### ISSUE THREAD

**Issue by sanjeevsuri** [tableau/TabMon#178]
Hello !

I am fairly new to this environment...so, please bear with me :)
I installed TabMon on one of our distributed(clustered) Tableau Dev servers (Windows 10) and could get through the PostgresSQL (TabMon) connection but the PostgresSQL (Server Auditing) connection is failing.
Server : localhost, Port: 8060
Database: workgroup
Username: readonly
Password: readonly

I did create the user prior to installing TabMon and restarted the services too
 tabadmin dbpass --username readonly readonly

The error I am getting is "could not connect to server: Connection refused (0x0000274D/10061) Is the server running on host "localhost" (::1) and accepting TCP/IP connections on port 8060? could not connect to server: Connection refused (0x0000274D/10061) Is the server running on host "localhost" (127.0.0.1) and accepting TCP/IP connections on port 8060?"

I did follow the leads online and on Github pertaining to modifying the pg_hba.conf and postgres.sql files under the C:\ProgramData\Tableau\Tableau Server\data\tabsvc\config but am just not able to identify the issue.

Under Local Connections, I have these entries.
host    all         rails         127.0.0.1/32          md5
host    all         tblwgadmin         127.0.0.1/32          md5
host    all         tableau     127.0.0.1/32          md5
host    all         readonly     127.0.0.1/32          md5
host    all         all         127.0.0.1/32          md5
host    all         rails         0:0:0:0:0:0:0:1/128          md5
host    all         tblwgadmin         0:0:0:0:0:0:0:1/128          md5
host    all         tableau     0:0:0:0:0:0:0:1/128          md5
host    all         readonly     0:0:0:0:0:0:0:1/128          md5
host    all         all         0:0:0:0:0:0:0:1/128          md5

Included the following line in postgressql.conf
listen_addresses = '*'

Any guidance is appreciated.

Best
Sanjeev

**Comment by sanjeevsuri** [comment]
Thanks for the response. 
TabMon is installed on the same system as the Tableau server and yes, it is otherwise functioning correctly. I can access the other dashboards in TabMon except "What's going on Dashboard?"
I'll double check the firewall rules but am doubtful that we have anything inhibiting those connections but I might be wrong. Do I need to check port 8060 in particular ?

Thanks again for getting back. Appreciate it.

**Comment by slewt** [comment]
Correct, it is TCP 8060 and should be from 127.0.0.1 -> 127.0.0.1 and/or ::1 -> ::1 given the way I am reading your current config correctly and the report is already published to Tableau Server, or you are using Tableau Desktop installed on the same server.

FYI: When you start to actually get a connection to the server, you should either connect successfully or your errors will change to something like:
'psql: FATAL:  password authentication failed for user readonly' or
'psql: FATAL: no pg_hba.conf entry for host "<IP Address>", user "readonly", database "workgroup", SSL off'

**Comment by sanjeevsuri** [comment]
As a matter of fact I did install Tableau Desktop on the same system where I have Tableau Server installed.
So, there might be 3 instances of Postgres. 
So I have TabMon, Tableau Desktop and Tableau Server, all installed on the same server.

**Comment by sanjeevsuri** [comment]
Just a quick update. Since we are in a clustered environment, I figured the postgres repository sits on one of the worker nodes. In order to access "What's going on" Dashboard, the connection had to be directed to the worker node instead of the primary node. That resolved the issue.

**Comment by danjrahm** [comment]
Hello,

Thanks for the update! I'm glad you were able to resolve the issue. If you have any other questions or issues please let me know!

Thanks,
Dan
